docs/docs/guides/frameworks/creating-a-framework-integration.mdx

```diff
@@ -0,0 +1,49 @@
+
+The core functionalities of Auth.js - `@auth/core` - are built on top of the Web Standard [Request](https://developer.mozilla.org/en-US/docs/Web/API/Request)/[Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) mental model, and therefore are framework-agnostic. For each framework, we provide an integration layer that allows you to use the authentication features in a way that is specific to the framework. See the [list of integrations](./reference#integrations) that are currently available.
+
+We welcome contributions of new official integrations. If you are interested in creating & maintaining a new integration, please read the following guidelines.
+
+## Official framework guidelines
+
+If you want to create a new official framework integration and distribute it under `@auth/` namespace, please make sure to follow the next steps and fulfill our maintenance requirements below:
+
+### Setting up a new integration
+
+We provide a script that generates all the required files for a new integration. To run the script, run the following command:
+
+```bash
+pnpm setup-fw-integration <framework-name>
+```
+
+This will copy all the files from our official template at `./packages/frameworks-template` to new directory under `packages/` with all the required files, and rename the placeholders to the name of the framework you provided.
+
+### Checklist for contributor
+
+- Coding styles. The source code files should:
+
+  - Be written in TypeScript
+  - Pass the linting rules of the monorepo
+  - Have a named export exported from its main module. For example: `export function ExpressAuth()`
+
+- API reference documentation - We use [TypeDoc](https://typedoc.org/) for automated documentation generation. The documentation should:
+
+  - Be written in JSDoc comments.
+  - Explain how to use and configure the integration: How to do Session management, how to login/logout, how to configure the base URL.
+  - Include a link to the official framework documentation in the reference section.
+
+- Deployed Example. This task can be in a follow-up PR. The deployed example should:
+
+  - All the actions (URL) should work as expected.
+  - Have at least one OAuth provider configured.
+  - The example code should live under `apps/examples/<framework-name>`. For example: `apps/examples/express`.
+
+- Miscellaneous:
+
+  - Add a redirect rule for the documentation page. See [this example](https://github.com/nextauthjs/next-auth/blob/e34d7ad5f2eb6129a5d313cbcfb593eb94d59857/docs/vercel.json#L75-L79) of a redirect rule for Express
+  - Add an entry to our manifest generation. See [this example](https://github.com/nextauthjs/next-auth/blob/e34d7ad5f2eb6129a5d313cbcfb593eb94d59857/docs/scripts/generate-manifest.mjs#L58-L65) of a manifest entry for Express.
+  - Add an entry to the PR labeler. See [this example](https://github.com/nextauthjs/next-auth/blob/46264fb42af4c3ef7137a5694875eaa1309462ea/.github/pr-labeler.yml#L30) of a labeler entry for SvelteKit.
```

Comment:
```suggestion
```

These can be part of the `setup-fw-integration` script to further reduce friction.